Quarkus should be able to run tests in different folders than the "preset" ones

**Description**
This issue comes from: [here](https://stackoverflow.com/questions/61299474/how-do-i-configure-quarkus-to-run-tests-in-folders-other-than-test-and-native-te)
I want to separate my unit tests from my integration tests for multiple reasons. To do that, I can add a set sourceSet (in gradle) or similar in maven, but when running the tests, I get this message:

> Caused by: java.lang.RuntimeException: The test class class YourTest is not located in any of the directories [classes/scala/native-test, classes/java/test, classes/scala/test, classes/java/native-test, /test-classes, bin/test, classes/kotlin/test, classes/kotlin/native-test]

It should be possible to run tests outside these folder.

**Implementation ideas**
I don't have any. But if you provide me some guidance, I would love to create the PR for that.

--- a/test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java
@@ -17,16 +17,22 @@ import io.quarkus.runtime.util.ClassPathUtils;
  */
 public final class PathTestHelper {
     private static final Map<String, String> TEST_TO_MAIN_DIR_FRAGMENTS = new HashMap<>();
+
     static {
-        // eclipse
+        //region Eclipse
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "bin" + File.separator + "test",
                 "bin" + File.separator + "main");
-        // idea
+        //endregion
+
+        //region Idea
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "out" + File.separator + "test",
                 "out" + File.separator + "production");
-        // gradle
+        //endregion
+
+        // region Gradle
+        // region Java
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "classes" + File.separator + "java" + File.separator + "native-test",
                 "classes" + File.separator + "java" + File.separator + "main");
@@ -34,22 +40,65 @@ public final class PathTestHelper {
                 "classes" + File.separator + "java" + File.separator + "test",
                 "classes" + File.separator + "java" + File.separator + "main");
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "java" + File.separator + "integration-test",
+                "classes" + File.separator + "java" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "java" + File.separator + "integrationTest",
+                "classes" + File.separator + "java" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "java" + File.separator + "native-integrationTest",
+                "classes" + File.separator + "java" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "java" + File.separator + "native-integration-test",
+                "classes" + File.separator + "java" + File.separator + "main");
+        //endregion
+        //region Kotlin
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "classes" + File.separator + "kotlin" + File.separator + "native-test",
                 "classes" + File.separator + "kotlin" + File.separator + "main");
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "classes" + File.separator + "kotlin" + File.separator + "test",
                 "classes" + File.separator + "kotlin" + File.separator + "main");
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "kotlin" + File.separator + "integration-test",
+                "classes" + File.separator + "kotlin" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "kotlin" + File.separator + "integrationTest",
+                "classes" + File.separator + "kotlin" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "kotlin" + File.separator + "native-integrationTest",
+                "classes" + File.separator + "kotlin" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "kotlin" + File.separator + "native-integration-test",
+                "classes" + File.separator + "kotlin" + File.separator + "main");
+        //endregion
+        //region Scala
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "classes" + File.separator + "scala" + File.separator + "native-test",
                 "classes" + File.separator + "scala" + File.separator + "main");
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 "classes" + File.separator + "scala" + File.separator + "test",
                 "classes" + File.separator + "scala" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "scala" + File.separator + "integration-test",
+                "classes" + File.separator + "scala" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "scala" + File.separator + "integrationTest",
+                "classes" + File.separator + "scala" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "scala" + File.separator + "native-integrationTest",
+                "classes" + File.separator + "scala" + File.separator + "main");
+        TEST_TO_MAIN_DIR_FRAGMENTS.put(
+                "classes" + File.separator + "scala" + File.separator + "native-integration-test",
+                "classes" + File.separator + "scala" + File.separator + "main");
+        //endregion
+        //endregion
 
-        // maven
+        //region Maven
         TEST_TO_MAIN_DIR_FRAGMENTS.put(
                 File.separator + "test-classes",
                 File.separator + "classes");
+        //endregion
     }
 
     private PathTestHelper() {

